### PR TITLE
os/board/rtl8721csm: add wpa tkip aes mixed mode in wifi

### PR DESCRIPTION
**Pull request by gexuyan**
Add the wpa tkip aes mixed security mode in wifi scan result

### CODE REVIEW

File: os/board/rtl8721csm/src/component/common/api/wifi/rtw_wpa_supplicant/wpa_supplicant/wifi_wps_config.c [hunk 1]
```diff
@@ -320,7 +320,11 @@ static void wps_config_wifi_setting(rtw_network_info_t *wifi, struct dev_credent
 		}
 	}
 	else if(dev_cred->auth_type & (WPS_AUTH_TYPE_WPA_PERSONAL | WPS_AUTH_TYPE_WPA_ENTERPRISE)) {
-		if(dev_cred->encr_type & WPS_ENCR_TYPE_AES) {
+		if((dev_cred->encr_type & WPS_ENCR_TYPE_AES) && (dev_cred->encr_type & WPS_ENCR_TYPE_TKIP)) {
+			printf("\r\nsecurity_type = RTW_SECURITY_WPA_MIXED_PSK\n");
```
Comment (pillip8282) [inline os/board/rtl8721csm/src/component/common/api/wifi/rtw_wpa_supplicant/wpa_supplicant/wifi_wps_config.c:324]: printf is not allowed in kernel space. using vdvdbg for information or vddbg for error

File: os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c [hunk 2]
```diff
@@ -774,7 +774,7 @@ static void print_scan_result(rtw_scan_result_t *record)
...
 	RTW_API_INFO(" %d\t  ", record->wps_type);
-	RTW_API_INFO("%s\t\t ", (record->security == RTW_SECURITY_OPEN) ? "Open" : (record->security == RTW_SECURITY_WEP_PSK) ? "WEP" : (record->security == RTW_SECURITY_WPA_TKIP_PSK) ? "WPA TKIP" : (record->security == RTW_SECURITY_WPA_AES_PSK) ? "WPA AES" : (record->security == RTW_SECURITY_WPA2_AES_PSK) ? "WPA2 AES" : (record->security == RTW_SECURITY_WPA2_TKIP_PSK) ? "WPA2 TKIP" : (record->security == RTW_SECURITY_WPA2_MIXED_PSK) ? "WPA2 Mixed" : (record->security == RTW_SECURITY_WPA_WPA2_MIXED) ? "WPA/WPA2 AES" : "Unknown");
+	RTW_API_INFO("%s\t\t ", (record->security == RTW_SECURITY_OPEN) ? "Open" : (record->security == RTW_SECURITY_WEP_PSK) ? "WEP" : (record->security == RTW_SECURITY_WPA_TKIP_PSK) ? "WPA TKIP" : (record->security == RTW_SECURITY_WPA_AES_PSK) ? "WPA AES" : (record->security == RTW_SECURITY_WPA_MIXED_PSK) ? "WPA Mixed" :(record->security == RTW_SECURITY_WPA2_AES_PSK) ? "WPA2 AES" : (record->security == RTW_SECURITY_WPA2_TKIP_PSK) ? "WPA2 TKIP" : (record->security == RTW_SECURITY_WPA2_MIXED_PSK) ? "WPA2 Mixed" : (record->security == RTW_SECURITY_WPA_WPA2_MIXED) ? "WPA/WPA2 AES" : "Unknown");
```
Comment (pillip8282) [inline os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c:777]: the line length is too long. please divide it.

File: os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c [hunk 3]
```diff
@@ -1001,7 +1001,7 @@ static void cmd_debug(int argc, char **argv)
 			RTW_API_INFO("BSSID : " MAC_FMT "\r\n", MAC_ARG(ap_info.BSSID.octet));
 			RTW_API_INFO("RSSI  : %d\r\n", ap_info.RSSI);
 			RTW_API_INFO("Beacon period : %d\r\n", ap_info.beacon_period);
-			RTW_API_INFO("Security : %s\r\n", (sec == RTW_SECURITY_OPEN) ? "Open" : (sec == RTW_SECURITY_WEP_PSK) ? "WEP" : (sec == RTW_SECURITY_WPA_TKIP_PSK) ? "WPA TKIP" : (sec == RTW_SECURITY_WPA_AES_PSK) ? "WPA AES" : (sec == RTW_SECURITY_WPA2_AES_PSK) ? "WPA2 AES" : (sec == RTW_SECURITY_WPA2_TKIP_PSK) ? "WPA2 TKIP" : (sec == RTW_SECURITY_WPA2_MIXED_PSK) ? "WPA2 Mixed" : "Unknown");
+			RTW_API_INFO("Security : %s\r\n", (sec == RTW_SECURITY_OPEN) ? "Open" : (sec == RTW_SECURITY_WEP_PSK) ? "WEP" : (sec == RTW_SECURITY_WPA_TKIP_PSK) ? "WPA TKIP" : (sec == RTW_SECURITY_WPA_AES_PSK) ? "WPA AES" : (sec == RTW_SECURITY_WPA_MIXED_PSK) ? "WPA Mixed" : (sec == RTW_SECURITY_WPA2_AES_PSK) ? "WPA2 AES" : (sec == RTW_SECURITY_WPA2_TKIP_PSK) ? "WPA2 TKIP" : (sec == RTW_SECURITY_WPA2_MIXED_PSK) ? "WPA2 Mixed" : "Unknown");
```
Comment (pillip8282) [inline os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c:1004]: the line length is too long. please divide it.

**Review by pillip8282 — CHANGES_REQUESTED** [review]

**Comment by vibhor-m** [comment]
reference for my id

**Comment by gexuyan** [comment]
will raise a new pr for this issue.

**Pull request closed without merging**